File: runme/mcgen/local/datan/exasrc/e1tim.f
      PROGRAM E1TIM
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      DIMENSION Y(36),ETA(50),CONETA(50),A(21,6)
      DIMENSION ATA1(6,6),ATA1AT(6,21),SCRAT(6,6)
      DATA Y /38.7D0,50.3D0,45.6D0,46.4D0,43.7D0,42.0D0,
     +21.8D0,21.8D0,51.3D0,39.5D0,26.9D0,23.2D0,
     +19.8D0,24.4D0,17.1D0,29.3D0,43.0D0,35.9D0,
     +19.6D0,33.2D0,38.8D0,35.3D0,23.4D0,14.9D0,
     +15.3D0,17.7D0,16.5D0, 6.6D0, 9.5D0, 9.1D0,
     + 3.1D0, 9.3D0, 4.7D0, 6.1D0, 7.4D0,15.1D0/
C identify program to user
      WRITE(*,*)' Program E1TIM demonstrates use of TIMSER'
      WRITE(*,*)' '
C initialize
      N=36
      K=2
      L=2
      P=0.9
C perform time series analysis
      CALL TIMSER(Y,N,K,L,P,ETA,CONETA,A,ATA1,ATA1AT,SCRAT)
C output results
      WRITE(*,'(3(A,I3),A,F10.5)')' N = ',N,', K = ',K,', L = ',L,
     +', P = ',P
      WRITE(*,*)'  I   Y         ETA       CONETA'
      DO 10 I=-K+1,N+K,1
        IF(I.LT. 1.OR. I.GT.N) THEN
          WRITE(*,'(I4,10X,2F10.5)')I,ETA(I+K),CONETA(I+K)
        ELSE
          WRITE(*,'(I4,3F10.5)')I,Y(I),ETA(I+K),CONETA(I+K)
        END IF
10    CONTINUE
      END

File: runme/mcgen/local/datan/exasrc/e1tim.c
#include <stdio.h>
#include "datsrc.h"

main(void)
{
    /* Local variables */ /* Initialized data */
    static integer i, k, l, n;
    static double y[36] = { 38.7,50.3,45.6,46.4,43.7,42.,21.8,21.8,51.3,
	      39.5,26.9,23.2,19.8,24.4,17.1,29.3,43.,35.9,19.6,33.2,38.8,35.3,
	      23.4,14.9,15.3,17.7,16.5,6.6,9.5,9.1,3.1,9.3,4.7,6.1,7.4,15.1 },
           a[126] /* was [21][6] */, p, scrat[36] /* was [6][6] */,
           ata1at[126] /* was [6][21] */, coneta[50],
           eta[50], ata1[36] /* was [6][6] */;
    /* System generated locals */
    integer i__1;
/* identify program to user */
    printf("%s\n\n", "Program E1TIM demonstrates use of TIMSER.");
/* initialize */
    n = 36;
    k = 2;
    l = 2;
    p = .9f;
/* perform time series analysis */
    timser_(y, &n, &k, &l, &p, eta, coneta, a, ata1, ata1at, scrat);
/* output results */
    printf("%s%3li%s%3li%s%3li%s%10.5lf\n",
           "N = ", n, ", K = ", k, ", L = ", l, ", P = ", p);
    printf("%s\n", "  I   Y         ETA       CONETA");
    i__1 = n + k;
    for (i = -k + 1; i <= i__1; ++i) {
	if (i < 1 || i > n) {
            printf("%4li          %10.5lf%10.5lf\n",
                   i, eta[i+k-1], coneta[i+k-1]);
	} else {
            printf("%4li%10.5lf%10.5lf%10.5lf\n",
                   i, y[i-1], eta[i+k-1], coneta[i+k-1]);
	}
    }
    return 0;
} /* main */

